vector/src/main/java/im/vector/app/features/home/room/detail/composer/MessageComposerViewModel.kt

```diff
@@ -741,14 +745,32 @@ class MessageComposerViewModel @AssistedInject constructor(
         voiceMessageHelper.startOrPauseRecordingPlayback()
     }
 
-    private fun handleEndAllVoiceActions(deleteRecord: Boolean) {
-        voiceMessageHelper.stopAllVoiceActions(deleteRecord)
+    private fun handleEndAllVoiceActions(deleteRecord: Boolean): MultiPickerAudioType? {
+        return voiceMessageHelper.stopAllVoiceActions(deleteRecord)
+    }
+
+    private fun handleInitializeVoiceRecorder(attachmentData: ContentAttachmentData) {
+        voiceMessageHelper.initializeRecorder(room.roomId, attachmentData)
+        setState { copy(voiceRecordingUiState = VoiceMessageRecorderView.RecordingUiState.Draft) }
     }
 
     private fun handlePauseRecordingVoiceMessage() {
         voiceMessageHelper.pauseRecording()
     }
 
+    private fun handleEntersBackground(composerText: String) {
+        withState {
+            if (it.isVoiceRecording) {
+                handleEndAllVoiceActions(deleteRecord = false)?.toContentAttachmentData()?.let { voiceDraft ->
```

Comment:
in order to move all of the voice draft logic to the SDK, we'd also need to move the voice helpers, is that what you had in mind @bmarty ?

Reply:
Not really, I would like the SDK be able to manage the draft file storage itself, but not the logic. We can discuss this on Monday.

Reply:
I could imagine something like... 

```kotlin
override fun initializeRecord(roomId: String) {
  outputFile = draftService.readFile(key = roomId)
}

override fun startRecord(roomId: String) {
  outputFile = draftService.createFile(key = roomId)
}
```